### PR TITLE
Releasing versioned (base) images

### LINKED ISSUE #10478: Tag version for the container base image
**Overview of the Feature Request**

People are sometimes struggling with the base image when building the application image. When there are changes pushed to the unstable tag, sometimes people still use a cached version and run into problems.

**What existing behavior do you want changed?**

We should tag the base image with some versioning scheme and make the application image use the tag by default. That way people will receive newer base image versions automatically.

**Any open or closed issues related to this feature request?**

More discussion here: https://dataverse.zulipchat.com/#narrow/stream/375812-containers/topic/change.20version.20scheme.20base.20image.3F

### PR DESCRIPTION
**Pull request by poikilotherm**
**What this PR does / why we need it**:
We want to release versioned tags of the base image (and eventually the application images, too). 
This requires setting up different versioning within the POM files, a better maintenance workflow to keep images updated and coordination between connected workflows.

**Which issue(s) this PR closes**:

Closes #10478

**Special notes for your reviewer**:
Open TODOs to call this done:
- [x] Application image push workflow needs a tag on PRs, as it can't use the input var from workflow_call
- [x] Needs testing if it works with the PR comment (needs a PR within the fork to test this without merging here first)
- [x] Move Docker Hub image description README update for gdcc/base to maintenance workflow, also listing the supported tags
- [x] Finalize PR after all testing etc is done by switching all the action references and many other pieces to match the upstream IQSS (see TODO annotations in the code)
- [x] Add release note
- [ ] Optional: add path filtering as annotated to avoid duplicate runs and race conditions
- [ ] Optional: add app image builds directly into the maintenance workflow (might be a separate PR within scope of #10618 )

**Suggestions on how to test this**:
All testing can only be done in the fork, as most of the workflows must be on the default branch to test them.
This has been done thoroughly, but there may be other non-testable edge cases we will only trip over once merged.

**Does this PR introduce a user interface change? If mockups are available, please link/include them here**:
No

**Is there a release notes update needed for this change?**:
Probably.

**Additional documentation**:

grafik
grafik

Preview docs at https://dataverse-guide--10827.org.readthedocs.build/en/10827/container/base-image.html

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -999,7 +999,9 @@
                 <app.image.tag>unstable</app.image.tag>
                 <app.skipDeploy>false</app.skipDeploy>
                 <base.image>gdcc/base:${base.image.tag}</base.image>
-                <base.image.tag>unstable</base.image.tag>
+                <base.image.flavor>noble</base.image.flavor>
```
Comment (pdurbin) [inline pom.xml:1002]:
```suggestion
                <!-- Flavor is a release from a Linux distro, such as Ubuntu -->
                <base.image.flavor>noble</base.image.flavor>
```

**Review by pdurbin — COMMENTED** [review]
Looks good to me! Just a tiny comment below.

**Review by pdurbin — APPROVED** [review]
I don't expect this to cause any problems for Dataverse itself (no Java code was touched). I'm merging this in the interest of science and progress. We can back it out if need be. 🚀
